cmd/gossamer/main.go

```diff
@@ -224,3 +237,55 @@ func initAction(ctx *cli.Context) error {
 
 	return nil
 }
+
+func buildSpecAction(ctx *cli.Context) error {
+	// set logger to critical, so output only contains genesis data
+	err := ctx.Set("log", "crit")
+	if err != nil {
+		return err
+	}
+	_, err = setupLogger(ctx)
+	if err != nil {
+		return err
+	}
+
+	var bs *dot.BuildSpec
+	if genesis := ctx.String(GenesisFlag.Name); genesis != "" {
+		bspec, e := dot.BuildFromGenesis(genesis)
+		if e != nil {
+			return e
+		}
+		bs = bspec
+	} else {
+		cfg, e := createBuildSpecConfig(ctx)
+		if e != nil {
+			return e
+		}
+		// expand data directory and update node configuration (performed separately
+		// from createDotConfig because dot config should not include expanded path)
+		cfg.Global.BasePath = utils.ExpandDir(cfg.Global.BasePath)
+
+		bspec, e := dot.BuildFromDB(cfg.Global.BasePath)
+		if e != nil {
+			return e
+		}
+		bs = bspec
+	}
+
+	if bs == nil {
+		return fmt.Errorf("error building genesis")
+	}
+
+	res := []byte{}  //nolint
+	if ctx.Bool(RawFlag.Name) {
```

Comment:
not sure i see the point of the raw flag, it should be turns it into raw every time? otherwise it's just copying the same file? maybe i'm missing something

Reply:
when you run it with no flags it output a human-readable JSON based on the genesis that's currently in the db created from the init command.  So, this is useful for getting an human-readable template for editing later.  Then use the `--raw` with `--genesis` to generate a raw output based on the genesis you passed in.

Reply:
i'm slightly confused, what db is it reading from?  it should be reading from the default genesis file if no genesis is provided

Reply:
what happens if there is no db? it's likely that the user will be doing `build-spec` before `init` since `init` requires the output of `build-spec`